tiledb/sm/array_schema/domain.cc

```diff
@@ -201,11 +201,7 @@ uint64_t Domain::cell_num_per_tile() const {
 
 template <>
 int Domain::cell_order_cmp_impl<char>(
-    const Dimension* dim, const UntypedDatumView a, const UntypedDatumView b) {
-  // Must be var-sized
-  assert(dim->var_size());
```

Comment:
Not a big deal as I'm planning to change a lot of these <char> implementations to <std::string> to prevent exactly this soon. This check might prevent someone using this function to process a CHAR datatype.

Reply:
In this case this check was not only unnecessary but not even the quite the right check.

Reply:
What would be the right check?